app/src/main/java/com/benoitquenaudon/daggerbundleinjectiondemo/BundleService.java

```diff
@@ -1,30 +1,26 @@
 package com.benoitquenaudon.daggerbundleinjectiondemo;
 
 import android.os.Bundle;
+import com.benoitquenaudon.daggerbundleinjectiondemo.injection.scope.ScopeActivity;
+import com.benoitquenaudon.daggerbundleinjectiondemo.ui.BaseAppCompatActivity;
+import javax.inject.Inject;
 
-public class BundleService {
-  private final Bundle data;
+@ScopeActivity public class BundleService {
+  private final Bundle all = new Bundle();
 
-  public BundleService(Bundle savedState, Bundle intentExtras) {
-    data = new Bundle();
-
-    if (savedState != null) {
-      data.putAll(savedState);
-    }
-    if (intentExtras != null) {
-      data.putAll(intentExtras);
-    }
+  @Inject BundleService(BaseAppCompatActivity activity) {
```

Comment:
I think it's better for the bundle service not to know anything about any `activity`.